### PR TITLE
Add Reactive Clear-Site-Data Support

### PR DESCRIPTION
**Pull request by sayembd**
1. A new implementation of ServerHttpHeadersWriter has been created to add Clear-Site-Data header support.
2. A new implementation of ServerLogoutHandler has been created which can be configured to write response headers during logout.
3. Added unit tests for both implementations.

### CODE REVIEW

File: web/src/main/java/org/springframework/security/web/server/authentication/logout/HeaderWriterServerLogoutHandler.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	private final ServerHttpHeadersWriter headersWriter;
+
+	/**
+	 * <p>Constructs a new instance using the {@link ServerHttpHeadersWriter} implementation.</p>
+
+	 * @param headersWriter a {@link ServerHttpHeadersWriter} implementation
+	 * @throws IllegalArgumentException if the argument is null
+	 */
+	public HeaderWriterServerLogoutHandler(ServerHttpHeadersWriter headersWriter) {
+		Assert.notNull(headersWriter, "ServerHttpHeadersWriter must not be null");
```
Comment (rhamedy) [inline web/src/main/java/org/springframework/security/web/server/authentication/logout/HeaderWriterServerLogoutHandler.java]: Curious if we should keep it `consistent` with the [servlet version](https://github.com/spring-projects/spring-security/blob/4a6e91531afb41392a44b9eb03d0636505a486fb/web/src/main/java/org/springframework/security/web/authentication/logout/HeaderWriterLogoutHandler.java#L41) which is `headersWriter cannot be null`   🤔

Comment (sayembd) [reply web/src/main/java/org/springframework/security/web/server/authentication/logout/HeaderWriterServerLogoutHandler.java]: Hey thanks for the review. Yeah it makes sense to keep it consistent across both version, at least the style of the message. I've forced-pushed the consistent version.

File: web/src/test/java/org/springframework/security/web/server/authentication/logout/HeaderWriterServerLogoutHandlerTests.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+				.isThrownBy(() -> new HeaderWriterServerLogoutHandler(null));
+	}
+
+	@Test
+	public void logoutWhenInvokedThenWritesResponseHeaders() {
+		ServerHttpHeadersWriter headersWriter = Mockito.mock(ServerHttpHeadersWriter.class);
+		HeaderWriterServerLogoutHandler handler = new HeaderWriterServerLogoutHandler(headersWriter);
+		ServerWebExchange serverWebExchange = Mockito.mock(ServerWebExchange.class);
+		WebFilterExchange filterExchange = Mockito.mock(WebFilterExchange.class);
+		Mockito.when(filterExchange.getExchange()).thenReturn(serverWebExchange);
```
[the remark this answers is not in the document]
Comment (sayembd) [reply web/src/test/java/org/springframework/security/web/server/authentication/logout/HeaderWriterServerLogoutHandlerTests.java]: Thanks for reviewing! I've just pushed the fix!

File: web/src/main/java/org/springframework/security/web/server/header/ClearSiteDataServerHttpHeadersWriter.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+	private final StaticServerHttpHeadersWriter headerWriterDelegate;
+
+	/**
+	 * <p>Constructs a new instance using the given directives.</p>
+	 *
+	 * @param directives directives that will be written as the header value
+	 * @throws IllegalArgumentException if the argument is null or empty
+	 */
+	public ClearSiteDataServerHttpHeadersWriter(Directive... directives) {
+		Assert.notEmpty(directives, "Directives cannot be empty or null.");
```
[the remark this answers is not in the document]
Comment (sayembd) [reply web/src/main/java/org/springframework/security/web/server/header/ClearSiteDataServerHttpHeadersWriter.java]: Thanks for reviewing! I've just pushed the fix!

**Comment by sayembd** [comment]
@jzheaux Thank you very much for the review! I've made the necessary adjustments and pushed the changes.

**Pull request merged**